### DISCUSSION TITLE
How to know if the widget is currently showing the upload list or the cloud image editor?

### DISCUSSION THREAD

**Discussion by martinsuhendra** [uploadcare/file-uploader#585] [Q&A]
Hi Uploadcare team,

Once we upload the image and the Edit image using cloud image editor button is shown, is there any way to know if we are on the upload list UI or we are on the cloud image editor UI?

I need a current state of the component if it's on the cloud image editor page or not.

<img width="853" alt="Screenshot 2024-01-05 at 16 19 57" src="https://github.com/uploadcare/blocks/assets/45756053/916fbc5b-5c9b-449c-b882-252dcd61904b">

Right now it shown on both UI, i can't determine if we are on editor UI or in upload list UI:

<img width="858" alt="Screenshot 2024-01-05 at 16 23 42" src="https://github.com/uploadcare/blocks/assets/45756053/2e2e93b0-4cb8-4795-bf71-c577c4973ac5">

Thank you!

**Comment by nd0ut** [comment]
Hey @martinsuhendra, thanks for inquiry.
We're working on events rework [here](https://github.com/uploadcare/blocks/pull/580) and there will be available event called `activity-change` which will be fired when the current activity (view) changed. But it's still needing some internal testing before release. So as a workaround, I could propose the following way: 

[![Edit File Uploader handle activity changes](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/file-uploader-regular-demo-forked-837lpj)
